modules/javafx.base/src/test/java/test/javafx/util/OutputRedirect.java

```diff
@@ -0,0 +1,232 @@
+    ///
+    public static void checkStderr(Object ... expected) {
+        if (stderrCapture != null) {
+            boolean err = false;
+            String text = stderrCapture.getAccumulatedOutput();
+
+            // exceptions
+            Map<String, Integer> errors = findErrors(text);
+            Map<String, Integer> exp = toMap(expected);
+            if (!errors.equals(exp)) {
+                stderr.println("Mismatch in thrown exceptions:\n  expected=" + exp + "\n  observed=" + errors);
+                err = true;
+            }
+
+            // patterns
+            for (Object x : expected) {
+                if (x instanceof String s) {
+                    if (!text.contains(s)) {
+                        stderr.println("Expected pattern not found: " + s);
+                        err = true;
+                    }
+                }
+            }
+
+            if (err) {
+                stderr.println(text);
+                // mismatch fails the test
+                fail("Unexpected stderr output");
+            }
+        }
+    }
+
+    /// Checks the accumulated stderr buffer for the expected exceptions and string patterns,
+    /// then restores the redirection.
+    ///
+    /// This method is equivalent to calling {@link #checkStderr(Object...)} followed by
+    /// {@link #restoreStderr()}.
+    ///
+    /// @param expected the expected exception classes (duplicates allowed), and/or string patterns
+    ///
+    public static void checkAndRestoreStderr(Object ... expected) {
+        try {
+            checkStderr(expected);
+        } finally {
+            restoreStderr();
+        }
+    }
+
+    private static Map<String, Integer> toMap(Object... expected) {
+        HashMap<String, Integer> m = new HashMap<>();
+        for (Object x : expected) {
+            if (x instanceof Class c) {
+                if (Throwable.class.isAssignableFrom(c)) {
+                    String name = c.getName();
+                    Integer v = m.get(name);
+                    if (v == null) {
+                        m.put(name, Integer.valueOf(1));
+                    } else {
+                        m.put(name, Integer.valueOf(v + 1));
+                    }
+                } else {
+                    throw new IllegalArgumentException("must specify Class<? extends Throwable>: " + c);
+                }
+            } else if(x instanceof String) {
+                // ok
+            } else {
+                throw new IllegalArgumentException("must specify either Class<? extends Throwable> or String: " + x);
+            }
+        }
+        return m;
+    }
+
+    private static Map<String, Integer> findErrors(String text) {
+        HashMap<String, Integer> m = new HashMap<>();
+        text.
+            lines().
+            map((s) -> findException(s)).
+            forEach((c) -> {
+                if (c != null) {
```

Comment:
Minor suggestion: filter for non-null Strings before the `forEach` rather than checking for non-null in the `forEach` ?

Reply:
my version is _more efficient_ :-)

Reply:
off topic, but it is more efficient - check out

https://github.com/andy-goryachev-oracle/Test/blob/main/src/goryachev/research/MoreEfficient.java